### ISSUE TITLE
Loading javascript quote model without window.checkoutConfig available causes js error

### ISSUE THREAD

**Issue by peterkeijsers** [buckaroo-it/Magento2#38]
The ApplePay module on the product page has a dependency on `Magento_Checkout/js/model/quote`. This model however needs the window.checkoutConfig object to be filled. The error on the product page is as follow:

`totals.js:16 Uncaught TypeError: Cannot read property 'items' of undefined
    at totals.js:16
    at Object.execCb (require.js:1650)
    at Object.context.execCb (resolver.js:145)
    at Module.check (require.js:866)
    at Module.enable (require.js:1143)
    at Module.init (require.js:774)
    at callGetModule (require.js:1170)
    at Object.completeLoad (require.js:1544)
    at HTMLScriptElement.onScriptLoad (require.js:1671)`

The problem lies in the `Magento_Checkout/js/model/quote` file on line 37 and 38
`totalsData = proceedTotalsData(window.checkoutConfig.totalsData),
        totals = ko.observable(totalsData),`

When these variables are not set, the `Magento_Checkout/js/model/totals` file crashes on line 16 `    var quoteItems = ko.observable(quote.totals().items),` because `quote.totals()` is undefined (empty still) because the window.checkoutConfig.totalsData is not set on product pages...

A quick fix for now is to remove the "buckaroo_magento2.product.info.applepay" block from the product page until the dependency issue is fixed.

https://github.com/buckaroo-it/Magento2/blob/144f38f3e948a0d33efb9b989c5d5810c3648de1/view/frontend/web/js/view/checkout/applepay/catalog-product-view.js#L23

**Comment by Buckaroo-Rens** [comment]
@peterkeijsers  Thank you for notifying us. We will look into this 'issue' later this week/beginning of next week.

**Comment by Buckaroo-Rens** [comment]
@peterkeijsers Thanks again for reaching out to us. We have tried to reproduce your issue but we cannot. Can you please provide us any additional information that can help us with researching this? Thank you in advance.

**Comment by peterkeijsers** [comment]
Hi @Buckaroo-Rens 

I can reproduce it on my fresh local Magento 2.3.5-p1 environment with Luma theme and buckaroo/magento2 v1.24.0.
I only have to install the module and enable Apple Pay method.

By enabling the module the following requirejs dependencies get called:

`Buckaroo_Magento2/js/view/checkout/applepay/catalog-product-view`
This module has the following dependencies:
```
- jquery
-- Magento_Ui/js/lib/core/collection 
--- Magento_Checkout/js/model/quote 
---- Buckaroo_Magento2/js/view/checkout/applepay/pay
```
And it this `Magento_Checkout/js/model/quote` dependency which eventually calls `totals.js` And in this file the quote has no totals property and thus causing the error.

I hope this helps in reproducing it.

**Comment by Buckaroo-Rens** [comment]
@peterkeijsers Thank you for the information. After testing with a clean, new installation we were able to reproduce your issue. We will solve this in our upcoming release. Thank you again for sharing.

**Comment by vladislav-padalka-hys** [comment]
FYI: should be gone in https://github.com/buckaroo-it/Magento2/releases/tag/v1.25.0-RC1

**Comment by Buckaroo-Rens** [comment]
Fix included in latest release 1.25.2
